Go tests fail on i686 and armv7 architectures

logs:
 
 - i686: https://kojipkgs.fedoraproject.org//work/tasks/7956/20857956/build.log
 - armv7: https://kojipkgs.fedoraproject.org//work/tasks/7958/20857958/build.log

It's the  same error repeated in each test:

```
+ go test -compiler gc -ldflags '' github.com/ncw/swift
Waiting for server to start . Started
2017/07/28 14:16:45 http: panic serving 127.0.0.1:60710: runtime error: invalid memory address or nil pointer dereference
goroutine 156 [running]:
net/http.(*conn).serve.func1(0x11a18180)
	/usr/lib/golang/src/net/http/server.go:1693 +0x9c
panic(0x2d2768, 0x49d178)
	/usr/lib/golang/src/runtime/panic.go:491 +0x204
sync/atomic.loadUint64(0x11a7410c, 0x119f6380, 0x11a68144)
	/usr/lib/golang/src/sync/atomic/64bit_arm.go:10 +0x3c
github.com/ncw/swift/swifttest.(*SwiftServer).serveHTTP(0x11a740f0, 0x472fd8, 0x11a20360, 0x119e6200)
	/builddir/build/BUILDROOT/golang-github-ncw-swift-0-0.8.gite3042b2.fc27.arm/usr/share/gocode/src/github.com/ncw/swift/swifttest/server.go:768 +0x90
github.com/ncw/swift/swifttest.NewSwiftServer.func1(0x472fd8, 0x11a20360, 0x119e6200)
	/builddir/build/BUILDROOT/golang-github-ncw-swift-0-0.8.gite3042b2.fc27.arm/usr/share/gocode/src/github.com/ncw/swift/swifttest/server.go:1093 +0x34
net/http.HandlerFunc.ServeHTTP(0x119ec050, 0x472fd8, 0x11a20360, 0x119e6200)
	/usr/lib/golang/src/net/http/server.go:1914 +0x34
net/http.serverHandler.ServeHTTP(0x11a04000, 0x472fd8, 0x11a20360, 0x119e6200)
	/usr/lib/golang/src/net/http/server.go:2610 +0x74
net/http.(*conn).serve(0x11a18180, 0x473528, 0x119f66c0)
	/usr/lib/golang/src/net/http/server.go:1797 +0x57c
created by net/http.(*Server).Serve
	/usr/lib/golang/src/net/http/server.go:2711 +0x208
--- FAIL: TestTransport (0.00s)
	swift_test.go:321: Auth failed Get http://127.0.0.1:33877/v1.0: EOF
```

The tests work fine in other arches like ppc64le or x86_64.

Any idea of what could be causing this?

--- a/swifttest/server.go
+++ b/swifttest/server.go
@@ -44,9 +44,12 @@ const (
 type HandlerOverrideFunc func(w http.ResponseWriter, r *http.Request, recorder *httptest.ResponseRecorder)
 
 type SwiftServer struct {
+	// `sync/atomic` expects the first word in an allocated struct to be 64-bit
+	// aligned on both ARM and x86-32.
+	// See https://golang.org/pkg/sync/atomic/#pkg-note-BUG for more details.
+	reqId    int64
 	sync.RWMutex
 	t        *testing.T
-	reqId    int64
 	mu       sync.Mutex
 	Listener net.Listener
 	AuthURL  string
@@ -121,12 +124,15 @@ type object struct {
 }
 
 type container struct {
+	// `sync/atomic` expects the first word in an allocated struct to be 64-bit
+	// aligned on both ARM and x86-32.
+	// See https://golang.org/pkg/sync/atomic/#pkg-note-BUG for more details.
+	bytes   int64
 	sync.RWMutex
 	metadata
 	name    string
 	ctime   time.Time
 	objects map[string]*object
-	bytes   int64
 }
 
 type segment struct {

--- a/swifttest/server.go
+++ b/swifttest/server.go
@@ -47,7 +47,7 @@ type SwiftServer struct {
 	// `sync/atomic` expects the first word in an allocated struct to be 64-bit
 	// aligned on both ARM and x86-32.
 	// See https://golang.org/pkg/sync/atomic/#pkg-note-BUG for more details.
-	reqId    int64
+	reqId int64
 	sync.RWMutex
 	t        *testing.T
 	mu       sync.Mutex
@@ -127,7 +127,7 @@ type container struct {
 	// `sync/atomic` expects the first word in an allocated struct to be 64-bit
 	// aligned on both ARM and x86-32.
 	// See https://golang.org/pkg/sync/atomic/#pkg-note-BUG for more details.
-	bytes   int64
+	bytes int64
 	sync.RWMutex
 	metadata
 	name    string

